mycpp/gc_list.h

```diff
@@ -154,7 +154,9 @@ template <typename T>
 void list_append(List<T>* self, T item) {
   StackRoots _roots({&self});
 
-  self->reserve(self->len_ + 1);
+  if (self->capacity_ - self->len_ < 1) {
```

Comment:
Why add these checks?  reserve() already has an if statement in it

I believe it matches how vector::reserve(n) works -- it's a no-op if it's already big enough

Reply:
Ah these were a bit preemptive. Should have read `reserve()` first. Will roll this back